CONTRIBUTING.md

```diff
@@ -281,11 +281,11 @@ nox -s pip-compile
 
 ## Releasing
 
-To release a new version of Cachi2, simply create a [GitHub release][cachi2-releases]. Note that
-Cachi2 follows [semantic versioning](https://semver.org/) rules.
+To release a new version of Cachi2, simply create a [GitHub release](https://github.com/hermetoproject/cachi2/releases).
+Note that Cachi2 follows [semantic versioning](https://semver.org/) rules.
```

Comment:
Never been a fan of hosting this maintainer information under CONTRIBUTING, but that's a fix for another day.